frontend/amundsen_application/static/js/ducks/lineage/index.spec.ts

```diff
@@ -0,0 +1,208 @@
+import { testSaga } from 'redux-saga-test-plan';
+
+import { Lineage } from 'interfaces';
+
+import * as API from './api/v0';
+
+import {
+  getTableLineage,
+  getTableLineageFailure,
+  getTableLineageSuccess,
+  getColumnLineage,
+  getColumnLineageFailure,
+  getColumnLineageSuccess,
+  getTableColumnLineageSuccess,
+  getTableColumnLineageFailure,
+  initialLineageState,
+} from './reducer';
+
+import {
+  getTableLineageWatcher,
+  getTableLineageWorker,
+  getColumnLineageWatcher,
+  getColumnLineageWorker,
+} from './sagas';
+
+import {
+  GetTableLineage,
+  GetColumnLineage,
+  GetTableColumnLineage,
+} from './types';
+
+describe('tableMetadata ducks', () => {
+  let testLineage: Lineage;
+
+  let columnName: string;
+  let testKey: string;
+
+  beforeAll(() => {
+    testKey = 'tableKey';
+    testLineage = {
+      upstream_entities: [
+        {
+          badges: [],
+          cluster: 'cluster',
+          database: 'database',
+          key: 'key',
+          level: 1,
+          name: 'name',
+          schema: 'schema',
+          usage: 100,
+          parent: 'parent',
+        },
+      ],
+      downstream_entities: [],
+      depth: 1,
+      direction: 'both',
+      key: testKey,
+    };
+
+    columnName = 'column_name';
+  });
+
+  describe('actions', () => {
+    it('getTableLineage - returns the action to get table lineage', () => {
+      const action = getTableLineage(testKey);
+      const { payload } = action;
+      expect(action.type).toBe(GetTableLineage.REQUEST);
+      expect(payload.key).toBe(testKey);
+    });
+
+    it('getTableLineage - returns the action to process failure', () => {
+      const status = 500;
+      const action = getTableLineageFailure(status);
+      const { payload } = action;
+      expect(action.type).toBe(GetTableLineage.FAILURE);
+      expect(payload.lineageTree).toBe(initialLineageState);
+      expect(payload.statusCode).toBe(status);
+    });
+
+    it('getTableLineage - returns the action to process success', () => {
+      const status = 200;
+      const action = getTableLineageSuccess(testLineage, status);
+      const { payload } = action;
+      expect(action.type).toBe(GetTableLineage.SUCCESS);
+      expect(payload.lineageTree).toBe(testLineage);
+      expect(payload.statusCode).toBe(status);
+    });
```

Comment:
Nit: I tend to group tests related to the same code by using a describe clause:
``
describe('getTableLineage', () => {
...
});
```